envoy/router/router.h

```diff
@@ -636,7 +637,7 @@ class VirtualHost {
   /**
    * @return const Config& the RouteConfiguration that owns this virtual host.
    */
-  virtual const Config& routeConfig() const PURE;
+  virtual const SharedConfig& routeConfig() const PURE;
```

Comment:
Shouldn't this return shared_ptr to avoid dangling lifetimes? See all my other comments about circular references though.

Reply:
I think it's unnecessary because the `SharedConfig` will always be valid for the VH.